### ISSUE TITLE
trapping errors in makenek

### ISSUE THREAD

**Issue by RonRahaman** [Nek5000/Nek5000#117]
Would you consider prefacing makenek with `set -e` and `set -o pipefail`?  

The problems are these lines at the end:

``` bash
make -j4 -f makefile 2>&1 | tee compiler.out
exit 0
```

This _always_ returns 0, even if makenek fails.  It's not very convenient if you're running makenek as part of a larger build script, since it doesn't trap any errors in makenek.

Using `set -e` exits on all errors.  Using `set -o pipefail` will trap any errors in `make` from in `make -j4 -f makefile 2>&1 | tee compiler.out`; without it, it will only trap errors from the last command in the pipe, which is `tee`.  

If you don't want to trap _all_ errors in makenek, then a more conservative option would be to use `set -o pipefail` and change the ending to:

``` bash
make -j4 -f makefile 2>&1 | tee compiler.out
exit $?
```

**Comment by stgeke** [comment]
Yes, that's a good idea. Do you want to create a PR for this?

**Comment by stgeke** [comment]
Fixed in #223
